### ISSUE TITLE
Investigate Historical Likelihood V1.1: margin uncertainty, scoring environment, and blowout saturation

### ISSUE THREAD

**Issue by sirgippy** [sirgippy/GippyRank4#26]
## Goal

Investigate whether GippyRank can improve its existing score-margin likelihood **without adding any new football statistic**.

The question is:

> **Are we extracting the right amount of evidence from the score margin we already observe?**

Historical Likelihood V1 has proven difficult to beat by adding box-score information. Before looking outward again, test whether its remaining weakness is the way it interprets the reliability and shape of margin evidence.

This investigation focuses on three hypotheses:

1. margin variance may depend on the expected mismatch between teams
2. total scoring environment may tell us how informative a given margin is
3. very large margins may contain diminishing exact information

This is a research investigation, not a production V1.1 implementation.

## Why

Current Historical Likelihood V1 already has a fairly rich **location/mean surface**:

- same-subdivision smooth rank-difference structure
- FBS/FCS-oriented cross-subdivision structure
- site effects
- 34 design features

But its residual distribution is comparatively simple:

- one global Student-t scale
- Student-t `df = 15`
- current df was selected from `{3, 5, 8, 15}` using 2003–2017 fitting and 2018–2021 development
- final V1 was refit through 2021

That implies several assumptions worth testing.

For example, V1 effectively treats the residual uncertainty around:

```text
10–7
31–28
63–60
```

similarly because each is a 3-point margin.

Likewise, it assumes one residual scale works for both near-even games and enormous expected mismatches.

And although Student-t tails make extreme outcomes robust, that is not the same as testing whether:

```text
42–0
56–0
70–0
```

should continue contributing increasingly precise information about relative quality.

## Core principle

Do **not** add new rating evidence in this investigation.

The observable football information remains:

```text
final score
site
pairing
```

Total points may be used only as **context for interpreting margin**, not as an independent quality factor.

Conceptually:

```text
p(margin | ranks, site, pairing, scoring_context)
```

is allowed.

This is not:

```text
p(total_points | ranks)
```

and total points must never receive its own direct rating factor.

## Start from current main

Before work:

```bash
git status
git fetch origin
git checkout main
git pull --ff-only origin main
```

Require a clean worktree and branch from the freshly updated `main`.

Use a branch such as:

```text
research/margin-likelihood-v1-1
```

## Frozen baseline

Historical Likelihood V1 is the baseline and must remain unchanged.

Preserve:

- current rank support
- current orientation semantics
- current FBS/FCS treatment
- current site semantics
- current rank-pair marginalization
- current training corpus
- current mean/design basis
- current production artifact
- current posterior behavior

Do not retune production V1.

For candidate models, keep the **same mean-feature family**. This investigation is about the observation distribution around that surface, not inventing a new rank relationship.

Use `df = 15` for all candidates so this experiment isolates the new assumptions rather than reopening the Student-t df search.

## Temporal design

Use:

```text
training:     2003–2017
development:  2018–2021
evaluation:   2022–2025
```

Candidate family, transforms, hyperparameter grids, and selection rules are frozen by this issue.

Use 2018–2021 only for candidate/hyperparameter selection.

After selection, refit the selected candidate using 2003–2021 before evaluating 2022–2025.

### Important caveat

2022–2025 has already been examined extensively by previous GippyRank research.

Therefore:

> **2022–2025 is not independent or untouched confirmation for this hypothesis.**

It may be used as leakage-safe historical evaluation, but never described as independent validation.

Do not use 2026 outcomes.

## Stage 0: diagnose V1 residual structure

Before fitting candidates, characterize V1 residual behavior using **training/development data only**.

Analyze residual magnitude and calibration against at least:

- absolute V1 expected-margin magnitude
- pairing:
  - FBS–FBS
  - FBS–FCS
  - FCS–FCS
- site where sample size permits
- total points
- observed margin magnitude

Use sensible fixed bins or quantiles derived from training data.

Report:

- residual MAE
- residual variance / robust scale
- 50%, 80%, and 95% empirical coverage
- sample counts
- whether scale appears meaningfully nonconstant

This diagnostic is descriptive. Do not use it to invent additional candidate families after seeing results.

## Candidate A: heteroskedastic margin scale

Keep the V1 location surface, but allow Student-t scale to vary with expected mismatch.

Use a positive log-scale model conceptually like:

```text
log(sigma)
    = pairing intercept
    + beta * log1p(abs(V1 expected margin))
```

Equivalent numerically stable parameterization is acceptable.

Requirements:

- mismatch must come from the rank/site hypothesis, not observed margin
- FBS/FCS orientation must remain correct
- scale must always be positive
- no observed box-score statistic enters Candidate A
- do not add arbitrary spline complexity

The purpose is narrowly:

> **Are some rank matchups intrinsically more or less variable than others?**

## Candidate B: total points as scale context

Extend Candidate A with total scoring environment:

```text
log(sigma)
    = Candidate A terms
    + gamma * standardized_log1p(total_points)
```

where:

```text
total_points = points_A + points_B
```

Centering/scaling parameters must be learned from training data only.

### Semantic requirement

Total points affects **scale only**.

It must not enter:

- the location/mean surface
- a direct quality factor
- a separate likelihood such as `p(total_points | ranks)`

Thus Candidate B asks:

> **Should a 3-point margin carry different confidence in a 17-point game than in a 123-point game?**

It does not ask whether high-scoring teams are better.

Add a rank-invariance/null test proving that total points alone cannot create quality evidence when all rank-dependent terms are neutralized.

## Candidate C: smooth blowout saturation

Extend Candidate B by testing whether extreme margins should have diminishing resolution.

Use the predeclared invertible transform:

```text
g_k(m) = k * asinh(m / k)
```

with:

```text
k ∈ {14, 21, 28, 42}
```

Properties:

- approximately linear near zero
- smoothly compresses large positive and negative margins
- remains symmetric
- does not impose a hard cap
- remains invertible

Fit the same V1 mean-feature family on the transformed response.

For direct game NLL comparisons on the original margin scale, include the appropriate change-of-variable Jacobian so Candidate C is scored comparably to raw-margin candidates.

Select `k` using development data only.

Do not introduce additional transform families after inspecting results.

The purpose is:

> **Does exact score differential become progressively less informative once a game is already a decisive blowout?**

## Candidate selection

Evaluate:

```text
V1
A
B
C(k=14)
C(k=21)
C(k=28)
C(k=42)
```

on 2018–2021 development.

Primary metric:

```text
marginalized whole-game NLL
```

Secondary metrics:

- expected-margin MAE
- 50% coverage
- 80% coverage
- 95% coverage
- interval widths where meaningful

Report aggregate and per-season results.

### Development pass gate

A candidate qualifies for posterior investigation if:

1. aggregate marginalized NLL improves over V1 by at least `0.005`
2. marginalized NLL improves in at least `3 of 4` development seasons
3. no development season worsens by more than `0.010` NLL
4. expected-margin MAE does not worsen by more than `0.10` points
5. absolute 80% coverage error does not worsen by more than `0.01`

A candidate may alternatively qualify on calibration if:

- aggregate NLL is within `0.002` of V1
- absolute 80% coverage error improves by at least `0.02`
- expected-margin MAE does not worsen by more than `0.10`

### Complexity preference

Among qualifying candidates, prefer the simplest.

Move from:

```text
V1 → A → B → C
```

only when the more complex candidate improves aggregate development NLL over the simpler qualifying candidate by at least `0.003`, or materially improves calibration under the alternative calibration rule above.

For Candidate C, choose exactly one `k` before 2022–2025 evaluation.

Record the selected candidate and all frozen parameters in a machine-readable model spec before final evaluation.

## Computational guardrail

**Do not run historical posterior panels for every candidate.**

Stage 1 should operate only on the historical likelihood dataset/rank-pair machinery and should be relatively cheap.

Only the **single development-selected candidate** may proceed to posterior evaluation.

If no candidate passes the development gate:

> stop the expensive investigation there

Generate the report/artifacts, open the research PR, and conclude that the new likelihood assumptions did not earn posterior evaluation.

Do not spend hours proving again that a development failure also fails inside BP.

## Rolling direct-likelihood robustness

For the selected candidate, if one exists, run a rolling historical **likelihood-level** robustness panel over earlier seasons where practical.

Each target season must be fit using only earlier seasons.

Do not reconstruct H/C priors or run posterior inference for this panel.

Report:

- target season
- V1 NLL
- candidate NLL
- delta
- coverage delta
- margin MAE delta

This panel is sensitivity evidence, not a separate hyperparameter-selection surface.

## Stage 2: posterior validation

Only if a candidate passes Stage 1, integrate that **one frozen candidate** into a research-only posterior path.

Do not replace production Historical Likelihood V1.

Evaluate 2022–2025 using identical:

- target seasons
- FBS teams
- game populations
- cutoffs
- rank support
- scoring keys

Run separately with:

```text
Context prior
History prior
```

Compare selected candidate vs production V1.

Report:

- final-rank NLL
- CRPS
- expected-rank MAE
- median-rank MAE
- 80% coverage
- 80% interval width
- Top 5 / Top 10 / Top 25 Brier
- per-season deltas

Use strict common keys.

## Posterior recommendation criteria

Because 2022–2025 is already familiar data, even a strong result is **not sufficient for immediate production promotion**.

Classify the result as:

### A — freeze as a Likelihood V1.1 candidate

Use when the selected model shows:

- average final-rank NLL improvement of at least `0.02`
- neither Context nor History aggregate NLL worsens
- improvement in at least `3 of 4` seasons for both prior families
- CRPS degradation no worse than `0.001`
- 80% coverage degradation no worse than `0.02`
- no catastrophic season

This means:

> promising enough to freeze for future independent confirmation

not:

> ship to production now

### B — useful likelihood insight, insufficient posterior gain

Use when the observation model measurably improves direct margin fit/calibration but does not produce stable enough ranking improvement.

### C — retain Historical Likelihood V1

Use when the new assumptions do not show meaningful stable benefit.

Do not force A.

## Optional future-game validation

Do not automatically run another expensive future-game experiment.

Run future-game validation only if the selected candidate reaches Recommendation A territory or comes within `0.005` aggregate final-rank NLL of that threshold while improving calibration.

If triggered, use the same frozen future-game scoring methodology and strict common keys used in prior research.

This validation is secondary and must not influence candidate selection.

## Current-2026 diagnostic

Do not use 2026 outcomes.

If and only if a candidate reaches Recommendation A, it is acceptable to generate a current 2026 **ranking-impact diagnostic** using games already present in the production snapshot.

Report largest expected-rank changes and uncertainty-width changes.

This is descriptive only and must not change production publication.

## Tests

Add focused semantic tests covering at least:

- Candidate A scale remains positive
- mismatch orientation is correct for FBS–FCS
- Candidate B total-points context enters scale only
- total points alone cannot create rank evidence
- training-derived total-points normalization does not use development/evaluation rows
- Candidate C transform is symmetric and monotonic
- Candidate C Jacobian is included in direct NLL scoring
- Candidate C transform approaches linear behavior near zero
- candidate selection reads only 2018–2021 development metrics
- 2022–2025 cannot affect candidate/hyperparameter selection
- strict comparison keys are identical
- production V1 artifacts remain unchanged

Do not add giant full-research builds to the ordinary unit-test path.

## Artifacts

Write deterministic research artifacts under something like:

```text
data/processed/margin_likelihood_v1_1/
```

Include at least:

```text
report.md
summary.json
model_spec.json
residual_diagnostics.csv
development_metrics.csv
development_season_metrics.csv
rolling_likelihood_metrics.csv
```

If Stage 2 runs, also include:

```text
posterior_metrics.csv
posterior_season_metrics.csv
calibration.csv
```

Plots are welcome when they materially clarify:

- residual scale vs expected mismatch
- calibration vs expected mismatch
- calibration vs total points
- behavior of the candidate margin transform

Do not generate decorative plots merely to fill a report.

## Scope / non-goals

Do not add:

- YPP
- yards
- plays
- turnovers
- first downs
- sacks
- EPA/PPA
- success rate
- play-by-play data
- offense/defense latent ratings
- time-varying team strength
- website changes
- new CFBD acquisition

Do not redesign the V1 mean surface.

Those are separate investigations.

This issue is intentionally narrow:

> **better interpretation of score margin, using only information already contained in the final score and existing game metadata**

## Validation

Run the ordinary project validation:

```bash
uv run pytest -q
uv run ruff check .
git diff --check
```

Run the research builder to completion for every stage actually triggered.

Verify deterministic substantive research artifacts with a repeat of the **cheap Stage 1 build**.

If Stage 2 is expensive, do not repeat a multi-hour posterior build solely to claim byte-for-byte determinism; clearly distinguish what was and was not replayed.

## Deliverables

Open a research PR and report on this issue:

- branch / PR / head commit
- exact frozen candidate specifications
- Stage 0 residual diagnostics
- 2018–2021 development metrics
- development-selected candidate, or explicit `none`
- why that selection occurred without using 2022–2025
- rolling likelihood robustness
- whether Stage 2 was triggered
- 2022–2025 posterior metrics if triggered
- calibration effects
- whether total-points context helped
- whether heteroskedasticity helped
- whether blowout compression helped
- final A/B/C recommendation
- explicit statement that 2022–2025 is not independent confirmation
- artifact footprint/runtime
- production V1 integrity confirmation
- tests/Ruff/diff-check
- PR status

The central question is:

> **Before asking the box score for more information, can GippyRank do a better job deciding how much to believe the scoreboard?**
